### PR TITLE
Support device_map for torch backend

### PR DESCRIPTION
**Pull request by fxmarty**
This is useful to benchmark large models on several GPUs.

Configs can include to set the number of devices to use.

```yml
hydra:
  job:
    env_set:
      CUDA_VISIBLE_DEVICES: 0,1
```

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
Thanks for addition

**Comment by IlyasMoutawwakil** [comment]
I guess there are some things that should be updated when `device_map` is used, so that the reported metrics are relevant.
Like memory consumption (it should include all devices) and maybe we need to synchronize all devices  for latency?

**Comment by fxmarty** [comment]
Right, good catch! I'll include that in this PR.

File: optimum_benchmark/main.py [hunk 1]
```diff
@@ -70,7 +70,6 @@ class ExperimentConfig:
             "cache_dir": None,
             "force_download": False,
             "local_files_only": False,
-            "use_auth_token": False,
```
Comment (fxmarty) [inline optimum_benchmark/main.py:73]: This argument is deprecated in transformers

Comment (IlyasMoutawwakil) [reply optimum_benchmark/main.py:73]: yes was it substituted with `token` argument ?

Comment (fxmarty) [reply optimum_benchmark/main.py:73]: Not necessarily, you can use `huggingface-cli login` and it works.

**Review by IlyasMoutawwakil — CHANGES_REQUESTED** [review]

File: optimum_benchmark/trackers/latency.py [hunk 2]
```diff
@@ -46,3 +46,41 @@ def _cpu_latency(self):
...
+        super().__init__(backend)
+        if backend.config.device_map:
+            self.hf_device_map = backend.pretrained_model.hf_device_map
+            self.end_device = max(self.hf_device_map.values())
+            self.device_indexes = set(self.hf_device_map.values())
+        else:
+            self.hf_device_map = None
+            self.end_device = self.device
+            if self.device.type == "cuda":
+                self.device_indexes = {self.device.index if self.device.index is not None else 0}
```
[the remark this answers is not in the document]
Comment (fxmarty) [reply optimum_benchmark/trackers/latency.py:61]: @IlyasMoutawwakil When using `CUDA_VISIBLE_DEVICES=3 python my_torch_script.py`, PyTorch remaps the available device to zero (i.e. `torch.cuda.current_device()` returns 0) so I think it is fine.

File: optimum_benchmark/utils.py [hunk 3]
```diff
@@ -27,7 +27,8 @@ def set_seed(seed: int) -> None:
 
 
 def bytes_to_mega_bytes(bytes: int) -> int:
-    return bytes >> 20
+    # Reference: https://en.wikipedia.org/wiki/Byte#Multiple-byte_units
+    return bytes * 1e-6
```
[the remark this answers is not in the document]
Comment (fxmarty) [reply optimum_benchmark/utils.py:31]: No. Should it?

File: optimum_benchmark/utils.py [hunk 4]
```diff
@@ -27,7 +27,8 @@ def set_seed(seed: int) -> None:
 
 
 def bytes_to_mega_bytes(bytes: int) -> int:
```
Comment (fxmarty) [inline optimum_benchmark/utils.py:29]:
```suggestion
def bytes_to_mega_bytes(bytes: int) -> float:
```

**Review by IlyasMoutawwakil — APPROVED** [review]

**Pull request merged**